design_documents/test_design.md

```diff
@@ -0,0 +1,30 @@
+# Test design
+
+## Context
+
+Current test design in Starknet Forge has some issues:
+- gas estimation is problematic since we have to distinguish between pure Cairo gas
+(e.g., creating an array in test) and Starknet gas
+- starknet functions like `get_caller_address` are unavailable in tests directly
+- starknet system calls like `library_call` are unavailable in tests directly
+- testing internal contract functions requires exposing them as `external`
```

Comment:
So it is just not possible to test internal at all.

Reply:
Discussed offline, will add an example